### PR TITLE
makes certain vamp powers work in containers

### LINKED ISSUE #9441: Can't use thrall speak in a pod
### Labels

[BUG]

### Description

Same as title

### Steps to reproduce

1. Unlock thralls
2. Get in a pod
3. Try to speak to thralls
4. You can't use this spell here

### Expected Behavior

to be able to speak to my thralls

### Additional Information & Screenshots

Reported by: The_Real_Mr_N
On server: Goonstation 2 Classic: Bombini
Round log date: 2022-07-02-18-54
Reported on: 2022-07-02 20:01:29

### PR DESCRIPTION
**Pull request by ZWRh3**
[BALANCE][BUG]
## About the PR 
makes some vamp powers such as cancel stuns and thrallspeak work in containers

## Why's this needed? 
fixes #9441, allows for vampires to use certain powers which shouldnt be restricted for being in a container such as cancel stuns

(not i am not aware of the balance implications of this and the port a brig)

### CODE REVIEW

**Comment by nefarious6th** [comment]
for balance in the pbrig perspective, most of these seem okay for both inside lockers and the pbrig; i think thrallspeak makes a lot of sense for either scenario, im not particularly drawn to radio jam needing to be outside-containers only, and the cancel stun for the sake of the pbrig doesn't really matter. enthralling from within containers seems weird for within the pbrig but it could probably lead to some funny moments and i see a lot of value in being able to enthrall from inside lockers/coffins.

coffin escape being usable in the pbrig is a huge detriment though because it guarantees there is no reliable way to catch a vampire for brigging, since they can bat out of normal brigs and that leaves the pbrig as an only resort for secure jailing. in a general sense, that would make me more likely to just outright hunt and kill vamps instead of trying to catch them and give appropriate sentences they could serve in the pbrig. likewise with screech inside of a pbrig; arcfiend's AOE disorient stun leaves me the same way when they're inside the pbrig; i'd much rather have executed a problematic arcfiend before shuttle arrival versus trying to take them along in the pbrig when they're just going to chain shock everyone. i see screech going similarly.

**Comment by ZWRh3** [comment]
should i add a 2 option of the var which is all containers but the port a brig?

**Comment by nefarious6th** [comment]
if you can make it work, then by all means; the variable could probably see some use on other antags' abilities later on down the line too. vampire coffin escaping from inside a locker or whatever seems neat to me

**Comment by ZWRh3** [comment]
> if you can make it work, then by all means; the variable could probably see some use on other antags' abilities later on down the line too. vampire coffin escaping from inside a locker or whatever seems neat to me

funnily enough i mainly did it out of remembering that one time i got validhunted using the fucking port a medbay because it preventing using cancel stuns

**Review by Tarmunora — CHANGES_REQUESTED** [review]
imo ditch the portabrig stuff. No need to treat it any differently than say, a welded locker. (iread: block coffin escape in all objects)

**Comment by ZWRh3** [comment]
> 

would screaming while in the port a brig be ok?

**Review by Tarmunora — APPROVED** [review]

File: code/datums/abilities/vampire/coffin.dm [hunk 1]
```diff
@@ -50,6 +50,7 @@
 	max_range = 999
 	cooldown = 600
 	pointCost = 0
+	not_when_in_an_object = 0
```
Comment (ZeWaka) [inline code/datums/abilities/vampire/coffin.dm:53]: FALSE

File: code/datums/abilities/vampire.dm [hunk 2]
```diff
@@ -438,6 +434,7 @@
 	preferred_holder_type = /datum/abilityHolder/vampire
 	var/when_stunned = 0 // 0: Never | 1: Ignore mob.stunned and mob.weakened | 2: Ignore all incapacitation vars
 	var/not_when_handcuffed = 0
+	var/not_when_in_an_object = 1
```
Comment (ZeWaka) [inline code/datums/abilities/vampire.dm:437]: TRUE

File: code/datums/abilities/vampiric_thrall.dm [hunk 3]
```diff
@@ -123,6 +123,7 @@
 	preferred_holder_type = /datum/abilityHolder/vampiric_thrall
 	var/when_stunned = 1 // 0: Never | 1: Ignore mob.stunned and mob.weakened | 2: Ignore all incapacitation vars
 	var/not_when_handcuffed = 0
+	var/not_when_in_an_object = 1
```
Comment (ZeWaka) [inline code/datums/abilities/vampiric_thrall.dm:126]: ditto everywhere else

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/datums/abilities/vampire/coffin.dm [hunk 4]
```diff
@@ -50,6 +50,7 @@
 	max_range = 999
 	cooldown = 600
 	pointCost = 0
+	not_when_in_an_object = FALSE
```
Comment (flrsh) [inline code/datums/abilities/vampire/coffin.dm:53]:
```suggestion

```

**Pull request merged**